src/main/java/uk/ac/ebi/ena/webin/cli/service/CompleteSampleService.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018-2021 EMBL - European Bioinformatics Institute
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package uk.ac.ebi.ena.webin.cli.service;
+
+import java.io.StringReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import uk.ac.ebi.ena.webin.cli.service.exception.ServiceMessage;
+import uk.ac.ebi.ena.webin.cli.service.exception.ServiceiException;
+import uk.ac.ebi.ena.webin.cli.service.handler.NotFoundErrorHandler;
+import uk.ac.ebi.ena.webin.cli.service.utils.HttpHeaderBuilder;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Attribute;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Sample;
+
+ /**
+    This class is used to get the complete sample i.e sampleXml + sampleResponceObject
+    For this it will call samples/{id} and cli/reference/sample/{id} APIs
+ */
+public class CompleteSampleService extends WebinService

```

Comment:
- We should move SampleService and SampleXmlService from webin-cli to webin-cli-validator just as they are.
- The getCompleteSample(String sampleValue ) and getSampleFromAPI are not generic methods and should not be in webin-cli-validator. Instead this code should be in sequencetools and should call a new service interface in the same package as SequenceRetrievalService:
 ```
/**
 * Retrieves samples from Webin.
 */
public interface SampleRetrievalService {
```
The SampleRetrievalService will be registered in SequenceToolsServices:
```
public class SequenceToolsServices {

    private final static AtomicReference<MasterEntryService> masterEntryService = new AtomicReference<>(new MasterEntryService());
    private final static AtomicReference<SequenceRetrievalService> sequenceRetrievalService = new AtomicReference<>();
    private final static AtomicReference<SampleRetrievalService> sampleRetrievalService = new AtomicReference<>();
...
    public static void init(SequenceRetrievalService sequenceRetrievalService) {
        SequenceToolsServices.sequenceRetrievalService.compareAndSet(null, sequenceRetrievalService);
    }
 public static void init(WebinSampleRetrievalService sampleRetrievalService) {
        SequenceToolsServices.sampleRetrievalService.compareAndSet(null, sampleRetrievalService);
    }

```
- Add WebinSampleRetrievalService implements SampleRetrievalService that will call the SampleService and SampleXmlService in webin-cli-validator.
- SequenceTools will implement WebinSampleRetrievalService because Webin-CLI will not use the SampleRetrievalService interface because it is defined in sequencetools and Webin-CLI does not directly import sequencetools.